cross_connect_client/models/cross_connect_server.py

```diff
@@ -0,0 +1,180 @@
+# Copyright 2024 Akretion (http://www.akretion.com).
+# @author Florian Mounier <florian.mounier@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from urllib.parse import urlparse
+
+import requests
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class CrossConnectServer(models.Model):
+    _name = "cross.connect.server"
+    _description = "Cross Connect Server"
+    _inherit = "server.env.mixin"
+
+    name = fields.Char(
+        required=True, compute="_compute_name", readonly=False, store=True
+    )
+    server_url = fields.Char(required=True)
+    api_key = fields.Char(
+        required=True,
+    )
+    group_ids = fields.One2many(
+        "res.groups",
+        inverse_name="cross_connect_server_id",
+        string="Cross Connect Server Groups",
+        readonly=True,
+    )
+    menu_id = fields.Many2one(
+        "ir.ui.menu",
+        string="Menu",
+        help="Menu to display the Cross Connect Server in the menu",
+        compute="_compute_menu_id",
+        store=True,
+    )
+    web_icon_data = fields.Binary(
+        compute="_compute_web_icon_data", inverse="_inverse_web_icon_data"
+    )
+
+    @api.depends("server_url")
+    def _compute_name(self):
+        for record in self:
+            if not record.name or record.name == "/":
+                try:
+                    record.name = urlparse(record.server_url).hostname
+                except Exception:
+                    record.name = "/"
+
+    @api.depends("name", "group_ids")
+    def _compute_menu_id(self):
+        for record in self:
+            if not record.group_ids:
+                if record.menu_id:
+                    if record.menu_id.action:
+                        record.menu_id.action.unlink()
+                    record.menu_id.unlink()
+                record.menu_id = False
+                continue
+
+            menu_groups = self.env.ref("base.group_system") | record.group_ids
```

Comment:
`self.env.ref("base.group_system")` should be removed IMO because, the group does not come from server so it will display the new app menu but won't allow you to access it.